### DISCUSSION TITLE
MCO support for STM32L0 series

### DISCUSSION THREAD

**Discussion by tzijnge** [zephyrproject-rtos/zephyr#55885] [Q&A] [answered]
I was trying to use the MCO clock output feature on a Nucleo L073 development board by selecting these configuration options
- CLOCK_STM32_MCO1_SRC_HSE
- CLOCK_STM32_MCO1_DIV

I also added an overlay to do the proper pinctrl configuration to get the clock output on PA8, but it didn't work.

Then I found out that there is no support for this feature for L0 series in clock_control/Kconfig.stm32. When I manually edit this file to include SOC_SERIES_STM32L0X in the right places, everything is working fine and I see the expected clock signal on PA8.

My questions are:

1. why is clock output for STM32 L0 not supported in Kconfig? It seems to be supported properly in the HAL. Did someone forget to add this or is there some underlying reason why it's not there?
2. Is it safe for me to patch the Kconfig file locally and use the MCO feature?
3. Is there any chance that the clock output feature will be supported for the L0 series in the (near) future?

**Comment by GeorgeCGV** [comment, marked as answer]
Hello, @tzijnge. The MCO is supported by a limited number of families.

* #7672
* #54018
* #49882 

The idea is to make it more generic and support more targets. That is being discussed in #31912.

> why is clock output for STM32 L0 not supported in Kconfig?

Nobody needed it or had time.

> Is it safe for me to patch the Kconfig file locally and use the MCO feature?

Yes. Had similar stuff done for H7. Alternatively, if there is desire and time #31912 can be tackled.

> Is there any chance that the clock output feature will be supported for the L0 series in the (near) future?

I am not sure if anyone works actively on #31912 at the moment.

**Reply by tzijnge** [reply]
Thanks
